Replace BUILD_CUDA option with USE_CUDA

jsandham opened the pull request.

YvanMokwinski reviewed (approved):
it's a homogenization but I feel USING_CUDA would have been a better name when reading the cmake scripts.

jsandham commented:
> it's a homogenization but I feel USING_CUDA would have been a better name when reading the cmake scripts.

Ya I think I am mostly ambivalent between the different options. I kind of think "BUILD_USING_CUDA" is the most descriptive. Anyways given that the other libraries are using "USE_CUDA" ill stick with that since the goal here is to get rocsparse standardized with the other libraries.

jsandham commented on CMakeLists.txt:
```diff
@@ -79,7 +79,7 @@ option(BUILD_SHARED_LIBS "Build hipSPARSE as a shared library" ON)
 option(BUILD_CLIENTS_SAMPLES "Build examples" ON)
 option(BUILD_VERBOSE "Output additional build information" OFF)
-option(BUILD_CUDA "Build hipSPARSE for CUDA" OFF)
+option(USE_CUDA "Build hipSPARSE for CUDA" OFF)
```
Modified description to "Build hipSPARSE using CUDA backend"

ntrost57 commented:
I think this is a major change with potential to break users build chains. We should post-pone removal of this option to next major release. Can we not use both option strings until then?

jsandham commented:
> I think this is a major change with potential to break users build chains. We should post-pone removal of this option to next major release. Can we not use both option strings until then?

Agreed. I can either delay this submission or just allow users to use either option for now

ntrost57 commented:
2nd option sounds good to me

TorreZuk commented:
I like the 2nd option too, perhaps actually print a cmake deprecated option message.   So if they use BUILD_USING_CUDA you just warn them the option is deprecated, use the newer form.

jsandham commented:
@ntrost57 @TorreZuk Ok I have modified CMakeList.txt to allow users to continue using the BUILD_CUDA option (and added a deprecation message). If you are happy with this ill submit.

ntrost57 reviewed (commented):
lgtm
